### PR TITLE
Removed XRSettings usage from HDRP. Protected XR usage in XRGraphicsConfig with ifdefs. 

### PR DESCRIPTION
**Pull request by nerites**
Also added a stereoRenderingPath method to XRGraphicsConfig. 

Pending Xbox build check from Sam Bickley. 
Tim- what other tests need to be run on this branch before PR approval?

### CODE REVIEW

**Comment by nerites** [comment]
Solved the issue where XRGraphicsConfig was blocking builds (StereoRenderingPath is only defined in UnityEditor).

This should be ready for merge into master at this point.

File: com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/Settings/FrameSettings.cs [hunk 1]
```diff
@@ -147,11 +146,7 @@ public static void InitializeFrameSettings(Camera camera, RenderPipelineSettings
             // Planar and real time cubemap doesn't need post process and render in FP16
             aggregate.enablePostprocess = camera.cameraType != CameraType.Reflection && srcFrameSettings.enablePostprocess;
 
-#if UNITY_SWITCH
-            aggregate.enableStereo = false;
-#else
-            aggregate.enableStereo = camera.cameraType != CameraType.Reflection && srcFrameSettings.enableStereo && XRSettings.isDeviceActive && (camera.stereoTargetEye == StereoTargetEyeMask.Both) && renderPipelineSettings.supportStereo;
-#endif
+            aggregate.enableStereo = camera.cameraType != CameraType.Reflection && srcFrameSettings.enableStereo && XRGraphicsConfig.enabled && (camera.stereoTargetEye == StereoTargetEyeMask.Both);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/Settings/FrameSettings.cs]: Please, continue to test && renderPipelineSettings.supportStereo.
Support may allow us to remove shader variant and save memory, so it need to be tested here.

File: com.unity.render-pipelines.core/CoreRP/Common/XRGraphicsConfig.cs [hunk 2]
```diff
@@ -1,7 +1,13 @@
 using System;
 using UnityEditor;
-using UnityEngine.XR;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.core/CoreRP/Common/XRGraphicsConfig.cs]: I think we should move the file into a CoreRP/XR folder and not in common. A bit too generic. What do you think?

File: com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/Settings/FrameSettings.cs (hunk 1 above)
Comment (nerites) [reply com.unity.render-pipelines.high-definition/HDRP/RenderPipeline/Settings/FrameSettings.cs]: Yes, it's read-only access to XRSettings.enabled, which can only be set outside of playmode.

**Comment by sebastienlagarde** [comment]
note: need to run katana before merging

**Comment by sebastienlagarde** [comment]
katana is green

**Comment by nerites** [comment]
Thanks for running that for me! Just to confirm: the mac failure is known?

**Comment by sebastienlagarde** [comment]
yes, this mac failure is not due to mac, it happen sometime.

**Pull request merged**